src/main/java/org/codehaus/mojo/taglist/FileAnalyser.java

```diff
@@ -162,16 +155,15 @@ public Collection<TagReport> execute() throws MavenReportException {
      * Gives the list of files to scan.
      *
      * @return a List of File objects.
-     * @throws MavenReportException the Maven report exception.
      */
-    private List<File> findFilesToScan() throws MavenReportException {
+    private List<File> findFilesToScan() {
         List<File> filesList = new ArrayList<>();
         try {
             for (String sourceDir : sourceDirs) {
                 filesList.addAll(FileUtils.getFiles(new File(sourceDir), includes, excludes));
             }
         } catch (IOException e) {
-            throw new MavenReportException("Error while trying to find the files to scan.", e);
+            throw new UncheckedIOException("Error while trying to find the files to scan.", e);
```

Comment:
Why not the regular one, I mean you need to propagate it anyway and present to the user...

Reply:
I need call it from `canGenerateReport` which doesn't have defined exception ....

Reply:
Alright, please add a TODO because in the next Doxia you can throw a checked exception.

Reply:
done